src/renderers/dom/stack/client/DOMChildrenOperations.js

```diff
@@ -246,6 +249,29 @@ var DOMChildrenOperations = {
     }
   },
 
+  /**
+   * Warns when React DOM is mutated by an external source.
+   *
+   * @return {object} observer instance with pre-defined options.
+   */
+  mutationObserver: function() {
+    if (MutationObserver) {
+      var config = {
+        childList: true,
+        subtree: true,
+      };
+
+      // eslint-disable-next-line no-inner-declarations
+      function callback(mutations) {
+        warning(false, 'Warning, an outside source is mutating the DOM');
+      }
+
+      var observer = new MutationObserver(callback);
+      var reactRoot = document.querySelector('[data-reactroot]');
```

Comment:
yeah, i should look for all react trees and run the observer on each of them; will fix.